packages/augur-sdk/src/state/db/ZeroXOrders.ts

```diff
@@ -200,6 +210,20 @@ export class ZeroXOrders extends AbstractTable {
       this.augur.events.emit('OrderEvent', {eventType: OrderEventType.Cancel, ...storedOrder});
       return false;
     };
+
+    const multiAssetData = defaultAbiCoder.decode(multiAssetDataAbi, `0x${storedOrder.signedOrder.makerAssetData.slice(10)}`);
+    const amounts = multiAssetData[0] as BigNumber[];
+    if (!amounts[0].eq(1)) return false;
+    if (!amounts[1].eq(0)) return false;
+    if (!amounts[2].eq(0)) return false;
```

Comment:
Check amounts length to avoid calling `.eq` on `undefined`?

Reply:
Actually, check amounts length to avoid extra multi transfers in there, like another multi-transfer that targets dai but with actual value?

Reply:
I think this is implicit in the overall length check isn't it?

Reply:
explained to me how it isn't! Will fix.

Reply:
Explained offline, but for others reading. If someone takes the tokenValues and maps it to the same offset as tokenIds, you just saved yourself 4 * 32bytes (length + 3 * uint256's), giving you plenty of space to do whatever you want to the abi decoded structure